src/core/platforms/platformutilities.cpp

```diff
@@ -183,14 +183,22 @@ bool PlatformUtilities::rmFile( const QString &filename ) const
   return file.remove( filename );
 }
 
-bool PlatformUtilities::renameFile( const QString &filename, const QString &newname ) const
+bool PlatformUtilities::renameFile( const QString &oldFilePath, const QString &newFilePath, bool overwrite ) const
 {
-  QFileInfo fi( newname );
+  QFileInfo fi( newFilePath );
+
+  // Insure the path exists
   QDir dir( fi.absolutePath() );
   dir.mkpath( fi.absolutePath() );
 
-  QFile file( filename );
-  return file.rename( newname );
+  // If the renamed file exists, overwrite
+  if ( fi.exists() && overwrite )
```

Comment:
What if the `oldFilePath == newFilePath`, then you will remove the file file to be renamed causing data loss?

Reply:
@suricactus , fair point, fixed.